stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -718,21 +718,48 @@ internal struct _ContiguousArrayBuffer<Element>: _ArrayBufferProtocol {
     _internalInvariant(bounds.upperBound <= count)
 
     let initializedCount = bounds.upperBound - bounds.lowerBound
-    target.initialize(
-      from: firstElementAddress + bounds.lowerBound, count: initializedCount)
+    
+    //Future: we could relax the count requirement, but we'd need to handle
+    //incompletely consuming the buffer rather than just setting it to the
+    //empty buffer singleton
+    if !_isPOD(Element.self) && self.count == initializedCount && isUniquelyReferenced() {
+      _ = beginCOWMutation()
+      target.initialize(
+        from: firstElementAddress + bounds.lowerBound, count: initializedCount)
+      _storage = _emptyArrayStorage
+      endCOWMutation()
+    } else {
+      target.initialize(
+        from: firstElementAddress + bounds.lowerBound, count: initializedCount)
+    }
     _fixLifetime(owner)
     return target + initializedCount
   }
 
   @inlinable
-  internal __consuming func _copyContents(
+  internal consuming func _copyContents(
     initializing buffer: UnsafeMutableBufferPointer<Element>
   ) -> (Iterator, UnsafeMutableBufferPointer<Element>.Index) {
     guard buffer.count > 0 else { return (makeIterator(), 0) }
+    
     let c = Swift.min(self.count, buffer.count)
-    buffer.baseAddress!.initialize(
-      from: firstElementAddress,
-      count: c)
+    //Future: we could relax the count requirement, but we'd need to handle
+    //incompletely consuming the buffer rather than just setting it to the
+    //empty buffer singleton
+    if !_isPOD(Element.self) && self.count == buffer.count && isUniquelyReferenced() {
```

Comment:
Hopefully the isPOD check should both reduce code size and avoid wasting time on pointless refcount checks, since moving a bunch of Ints isn't going to be any better than copying them.

Reply:
yes, it should